[Accelerator4][PostgreSQL] - NumberFormatException in date time value when sending event polling request

NumberFormatException in date time value when sending event polling request.

> [2025-05-26 10:30:26,067] [1a83b408-f27d-43cc-a252-78ebbb8327f8] ERROR {org.wso2.financial.services.accelerator.event.notifications.service.handler.DefaultEventPollingServiceHandler} - Error occurred while polling events org.wso2.financial.services.accelerator.event.notifications.service.exception.FSEventNotificationException: Error when retrieving FS event notifications.
> 	at org.wso2.financial.services.accelerator.event.notifications.service.EventPollingService.pollEvents(EventPollingService.java:153)
> 	at org.wso2.financial.services.accelerator.event.notifications.service.handler.DefaultEventPollingServiceHandler.pollEvents(DefaultEventPollingServiceHandler.java:99)
> 	at org.wso2.financial.services.accelerator.event.notifications.endpoint.api.EventPollingEndpoint.pollEvents(EventPollingEndpoint.java:118)
> 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
> 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
> 	at org.apache.cxf.service.invoker.AbstractInvoker.performInvocation(AbstractInvoker.java:179)
> 	at org.apache.cxf.service.invoker.AbstractInvoker.invoke(AbstractInvoker.java:96)
> 	at org.apache.cxf.jaxrs.JAXRSInvoker.invoke(JAXRSInvoker.java:201)
> 	at org.apache.cxf.jaxrs.JAXRSInvoker.invoke(JAXRSInvoker.java:104)
> 	at org.apache.cxf.interceptor.ServiceInvokerInterceptor$1.run(ServiceInvokerInterceptor.java:59)
> 	at org.apache.cxf.interceptor.ServiceInvokerInterceptor.handleMessage(ServiceInvokerInterceptor.java:96)
> 	at org.apache.cxf.phase.PhaseInterceptorChain.doIntercept(PhaseInterceptorChain.java:307)
> 	at org.apache.cxf.transport.ChainInitiationObserver.onMessage(ChainInitiationObserver.java:121)
> 	at org.apache.cxf.transport.http.AbstractHTTPDestination.invoke(AbstractHTTPDestination.java:265)
> 	at org.apache.cxf.transport.servlet.ServletController.invokeDestination(ServletController.java:234)
> 	at org.apache.cxf.transport.servlet.ServletController.invoke(ServletController.java:208)
> 	at org.apache.cxf.transport.servlet.ServletController.invoke(ServletController.java:160)
> 	at org.apache.cxf.transport.servlet.CXFNonSpringServlet.invoke(CXFNonSpringServlet.java:225)
> 	at org.apache.cxf.transport.servlet.AbstractHTTPServlet.handleRequest(AbstractHTTPServlet.java:304)
> 	at org.apache.cxf.transport.servlet.AbstractHTTPServlet.doPost(AbstractHTTPServlet.java:217)
> 	at javax.servlet.http.HttpServlet.service(HttpServlet.java:555)
> 	at org.apache.cxf.transport.servlet.AbstractHTTPServlet.service(AbstractHTTPServlet.java:279)
> 	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:199)
> 	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:144)
> 	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:51)
> 	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:168)
> 	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:144)
> 	at org.apache.catalina.filters.HttpHeaderSecurityFilter.doFilter(HttpHeaderSecurityFilter.java:129)
> 	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:168)
> 	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:144)
> 	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:168)
> 	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90)
> 	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:482)
> 	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:130)
> 	at org.wso2.carbon.identity.context.rewrite.valve.TenantContextRewriteValve.invoke(TenantContextRewriteValve.java:145)
> 	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93)
> 	at org.wso2.carbon.identity.context.rewrite.valve.OrganizationContextRewriteValve.invoke(OrganizationContextRewriteValve.java:123)
> 	at org.wso2.carbon.tomcat.ext.valves.SameSiteCookieValve.invoke(SameSiteCookieValve.java:38)
> 	at org.wso2.carbon.identity.cors.valve.CORSValve.invoke(CORSValve.java:83)
> 	at org.wso2.carbon.identity.authz.valve.AuthorizationValve.invoke(AuthorizationValve.java:157)
> 	at org.wso2.carbon.identity.auth.valve.AuthenticationValve.invoke(AuthenticationValve.java:153)
> 	at org.wso2.carbon.tomcat.ext.valves.CompositeValve.continueInvocation(CompositeValve.java:110)
> 	at org.wso2.carbon.tomcat.ext.valves.TomcatValveContainer.invokeValves(TomcatValveContainer.java:49)
> 	at org.wso2.carbon.tomcat.ext.valves.CompositeValve.invoke(CompositeValve.java:71)
> 	at org.wso2.carbon.tomcat.ext.valves.CarbonStuckThreadDetectionValve.invoke(CarbonStuckThreadDetectionValve.java:152)
> 	at org.wso2.carbon.extension.identity.x509Certificate.valve.X509CertificateAuthenticationValve.invoke(X509CertificateAuthenticationValve.java:59)
> 	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:660)
> 	at org.wso2.carbon.identity.core.context.valve.IdentityContextCreatorValve.invoke(IdentityContextCreatorValve.java:45)
> 	at org.wso2.carbon.tomcat.ext.valves.CarbonContextCreatorValve.invoke(CarbonContextCreatorValve.java:63)
> 	at org.wso2.carbon.tomcat.ext.valves.RequestEncodingValve.invoke(RequestEncodingValve.java:49)
> 	at org.wso2.carbon.tomcat.ext.valves.RequestCorrelationIdValve.invoke(RequestCorrelationIdValve.java:137)
> 	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
> 	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:346)
> 	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:396)
> 	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63)
> 	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:937)
> 	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1791)
> 	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52)
> 	at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1190)
> 	at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
> 	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:63)
> 	at java.base/java.lang.Thread.run(Thread.java:1583)
> Caused by: org.wso2.financial.services.accelerator.event.notifications.service.exception.FSEventNotificationException: Error occurred while retrieving notifications for client ID : 'U78e1avYnpqA5FnRhSNygGQaYv0a'.
> 	at org.wso2.financial.services.accelerator.event.notifications.service.dao.PostgreSqlEventNotificationDAOImpl.getNotificationsByClientIdAndStatus(PostgreSqlEventNotificationDAOImpl.java:106)
> 	at org.wso2.financial.services.accelerator.event.notifications.service.EventPollingService.pollEvents(EventPollingService.java:104)
> 	... 61 more
> Caused by: org.postgresql.util.PSQLException: Bad value for type timestamp/date/time: {1}
> 	at org.postgresql.jdbc.TimestampUtils.parseBackendTimestamp(TimestampUtils.java:362)
> 	at org.postgresql.jdbc.TimestampUtils.toTimestamp(TimestampUtils.java:394)
> 	at org.postgresql.jdbc.PgResultSet.getTimestamp(PgResultSet.java:628)
> 	at org.postgresql.jdbc.PgResultSet.getTimestamp(PgResultSet.java:2666)
> 	at org.wso2.financial.services.accelerator.event.notifications.service.dao.PostgreSqlEventNotificationDAOImpl.getNotificationsByClientIdAndStatus(PostgreSqlEventNotificationDAOImpl.java:84)
> 	... 62 more
> Caused by: java.lang.NumberFormatException: Expected time to be colon-separated, got ''
> 	at org.postgresql.jdbc.TimestampUtils.parseBackendTimestamp(TimestampUtils.java:264)
> 	... 66 more

Test Case Id: "Initial Event polling request with valid inputs"

DB Version: PostgreSQL 17.2

--- a/financial-services-accelerator/components/org.wso2.financial.services.accelerator.event.notifications.service/src/main/java/org/wso2/financial/services/accelerator/event/notifications/service/dao/PostgreSqlEventNotificationDAOImpl.java
+++ b/financial-services-accelerator/components/org.wso2.financial.services.accelerator.event.notifications.service/src/main/java/org/wso2/financial/services/accelerator/event/notifications/service/dao/PostgreSqlEventNotificationDAOImpl.java
@@ -32,6 +32,7 @@ import java.sql.Connection;
 import java.sql.PreparedStatement;
 import java.sql.ResultSet;
 import java.sql.SQLException;
+import java.sql.Timestamp;
 import java.util.ArrayList;
 import java.util.List;
 
@@ -72,7 +73,6 @@ public class PostgreSqlEventNotificationDAOImpl extends EventNotificationDAOImpl
                     //read event notifications from the result set
                     while (notificationResultSet.next()) {
                         Notification notification = new Notification();
-
                         notification.setNotificationId(notificationResultSet.getString
                                 (EventNotificationConstants.NOTIFICATION_ID));
                         notification.setClientId(notificationResultSet.getString
@@ -81,8 +81,10 @@ public class PostgreSqlEventNotificationDAOImpl extends EventNotificationDAOImpl
                                 (EventNotificationConstants.RESOURCE_ID));
                         notification.setStatus(notificationResultSet.getString
                                 (EventNotificationConstants.STATUS));
-                        notification.setUpdatedTimeStamp((notificationResultSet.getTimestamp(
-                                (EventNotificationConstants.UPDATED_TIMESTAMP)).getTime()));
+                        long updatedTimeInepochSeconds = notificationResultSet.getLong(EventNotificationConstants.
+                                UPDATED_TIMESTAMP);
+                        Timestamp updatedTimestamp = new Timestamp(updatedTimeInepochSeconds * 1000);
+                        notification.setUpdatedTimeStamp(updatedTimestamp.getTime());
 
                         notificationList.add(notification);
                     }

